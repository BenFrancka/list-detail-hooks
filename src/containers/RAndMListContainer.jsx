import React, { useState, useEffect } from 'react';
import ListOfCharacters from '../components/characters/ListOfCharacters';
import { fetchCharacterData } from '../services/RAndMApi';

const RAndMListContainer = () => {
  const [loading, setLoading] = useState(true);
  const [characters, setCharacters] = useState([]);

  useEffect(() => {
    fetchCharacterData()
      .then((characters) => setCharacters(characters))
      .finally(() => setLoading(false));
  }, []);

  if(loading) {
    return (
      <img
        src="https://thumbs.gfycat.com/SoftEarnestGyrfalcon-max-1mb.gif"
        alt="loading spinner"
      />
    );
  }
  return <ListOfCharacters characters={characters} />;
};

export default RAndMListContainer;
