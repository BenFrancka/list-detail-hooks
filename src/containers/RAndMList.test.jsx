import React from 'react';
import { render, screen } from '@testing-library/react';
import RAndMListContainer from './RAndMListContainer';
import { MemoryRouter } from 'react-router-dom';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import listData from '../fixtures/ListData.json';

const server = setupServer(
  rest.get('https://rickandmortyapi.com/api/character', (req, res, ctx) => {
    return res(ctx.json(listData));
  })
);

describe('RAndMListContainer', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('displays a list of Rick and Morty Characters', async () => {
    const { container } = render(
      <MemoryRouter>
        <RAndMListContainer />
      </MemoryRouter>
    );

    screen.getAllByAltText('loading spinner');

    const ul = await screen.findByRole('list', { name: 'character-list' });
    expect(ul).not.toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });
});
