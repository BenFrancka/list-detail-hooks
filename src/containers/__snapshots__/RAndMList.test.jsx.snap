// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RAndMListContainer displays a list of Rick and Morty Characters 1`] = `
<div>
  <ul
    aria-label="character-list"
  >
    <a
      href="/1"
    >
      <li>
        <figure>
          <img
            alt="Rick Sanchez"
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          />
          <figcaption>
            Rick Sanchez
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/2"
    >
      <li>
        <figure>
          <img
            alt="Morty Smith"
            src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
          />
          <figcaption>
            Morty Smith
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/3"
    >
      <li>
        <figure>
          <img
            alt="Summer Smith"
            src="https://rickandmortyapi.com/api/character/avatar/3.jpeg"
          />
          <figcaption>
            Summer Smith
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/4"
    >
      <li>
        <figure>
          <img
            alt="Beth Smith"
            src="https://rickandmortyapi.com/api/character/avatar/4.jpeg"
          />
          <figcaption>
            Beth Smith
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/5"
    >
      <li>
        <figure>
          <img
            alt="Jerry Smith"
            src="https://rickandmortyapi.com/api/character/avatar/5.jpeg"
          />
          <figcaption>
            Jerry Smith
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/6"
    >
      <li>
        <figure>
          <img
            alt="Abadango Cluster Princess"
            src="https://rickandmortyapi.com/api/character/avatar/6.jpeg"
          />
          <figcaption>
            Abadango Cluster Princess
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/7"
    >
      <li>
        <figure>
          <img
            alt="Abradolf Lincler"
            src="https://rickandmortyapi.com/api/character/avatar/7.jpeg"
          />
          <figcaption>
            Abradolf Lincler
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/8"
    >
      <li>
        <figure>
          <img
            alt="Adjudicator Rick"
            src="https://rickandmortyapi.com/api/character/avatar/8.jpeg"
          />
          <figcaption>
            Adjudicator Rick
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/9"
    >
      <li>
        <figure>
          <img
            alt="Agency Director"
            src="https://rickandmortyapi.com/api/character/avatar/9.jpeg"
          />
          <figcaption>
            Agency Director
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/10"
    >
      <li>
        <figure>
          <img
            alt="Alan Rails"
            src="https://rickandmortyapi.com/api/character/avatar/10.jpeg"
          />
          <figcaption>
            Alan Rails
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/11"
    >
      <li>
        <figure>
          <img
            alt="Albert Einstein"
            src="https://rickandmortyapi.com/api/character/avatar/11.jpeg"
          />
          <figcaption>
            Albert Einstein
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/12"
    >
      <li>
        <figure>
          <img
            alt="Alexander"
            src="https://rickandmortyapi.com/api/character/avatar/12.jpeg"
          />
          <figcaption>
            Alexander
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/13"
    >
      <li>
        <figure>
          <img
            alt="Alien Googah"
            src="https://rickandmortyapi.com/api/character/avatar/13.jpeg"
          />
          <figcaption>
            Alien Googah
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/14"
    >
      <li>
        <figure>
          <img
            alt="Alien Morty"
            src="https://rickandmortyapi.com/api/character/avatar/14.jpeg"
          />
          <figcaption>
            Alien Morty
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/15"
    >
      <li>
        <figure>
          <img
            alt="Alien Rick"
            src="https://rickandmortyapi.com/api/character/avatar/15.jpeg"
          />
          <figcaption>
            Alien Rick
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/16"
    >
      <li>
        <figure>
          <img
            alt="Amish Cyborg"
            src="https://rickandmortyapi.com/api/character/avatar/16.jpeg"
          />
          <figcaption>
            Amish Cyborg
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/17"
    >
      <li>
        <figure>
          <img
            alt="Annie"
            src="https://rickandmortyapi.com/api/character/avatar/17.jpeg"
          />
          <figcaption>
            Annie
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/18"
    >
      <li>
        <figure>
          <img
            alt="Antenna Morty"
            src="https://rickandmortyapi.com/api/character/avatar/18.jpeg"
          />
          <figcaption>
            Antenna Morty
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/19"
    >
      <li>
        <figure>
          <img
            alt="Antenna Rick"
            src="https://rickandmortyapi.com/api/character/avatar/19.jpeg"
          />
          <figcaption>
            Antenna Rick
          </figcaption>
        </figure>
      </li>
    </a>
    <a
      href="/20"
    >
      <li>
        <figure>
          <img
            alt="Ants in my Eyes Johnson"
            src="https://rickandmortyapi.com/api/character/avatar/20.jpeg"
          />
          <figcaption>
            Ants in my Eyes Johnson
          </figcaption>
        </figure>
      </li>
    </a>
  </ul>
</div>
`;
